PKGBUILD

```diff
@@ -0,0 +1,42 @@
+# Maintainer: Amin Vakil <info AT aminvakil DOT com>
+# Contributor: Jelle van der Waa <jelle@archlinux.org>
+# Contributor: Frederik Schwan <freswa at archlinux dot org>
+# Contributor: Sander Boom <sanderboom@gmail.com>
+
+_pkgname=ansible-lint
+pkgname=ansible-lint-test
+pkgver=5.0.7.r73.gfef55b6
```

Comment:
In final version we should generate this based on current version. Usually I use `python setup.py --version` in order to retrieve the version. This has the advantage of re-using the setuptools-scm versioning.

I think it would be very cool to have a `tox -e pkg-arch` command that does build the package locally as it would make it much easier for others to debug/fix it if gets broken.  Having extra logic inside GHA file does not help much as we cannot run the same commands locally easily.

Reply:
Alternatively you can add a playbooks that runs these commands, something similar with eco.yml which we already have.

Reply:
`pkgver` gets replaced using `python setup.py --version` now. (https://github.com/ansible-community/ansible-lint/pull/1645/commits/e6ad474e25753c8184c2f11f4b5c9809126d1475)

Reply:
I'm also working something like `eco.yml`, will push here when it's ready.

Reply:
`python setup.py --version` or running
```
import setuptools_scm
print (setuptools_scm.get_version())
```
as @webknjaz mentioned (https://github.com/ansible/pylibssh/blob/6de4512/.github/workflows/build-test-n-publish.yml#L141-L148)
is returning 5.0.8.dev98+g04eb0f4 instead of 5.1.0... something.
I guess that's because I've cloned the repository when 5.0.8 was about to come out or something like that.
What should I do about this?

Reply:
You are right, if you forget to pull last tags you endup with various versions, is not really a bug. I encounter this issue quite often.

I think that we should better use pre-build sdist or wheel and rely on the version from within them. We can assume that a distribution package is produced from these and not directly from the cloned code and avoid the versioning issue. This should also allow you to drop setuptools-scm from being a dependency for the distro package.

Reply:
Exactly. Dirty versions are also produced when you mutate some files on disk (which could be ignored with `git update-index --assume-unchanged` in CI).
But yes, sdist is usually considered a canonical source from which you're able to build anything you want.

Reply:
Very well, I need a couple of days to get familiar with the stuff you're talking about :) I haven't done it before.
@webknjaz Thanks for the links!

Reply:
By the way, you could get the same SCM output by just using `python -m setuptools_scm` but that's not what you need when using the sdist metadata version anyway.